daphne/src/roles_test.rs

```diff
@@ -499,6 +501,49 @@ async fn get_reports_empty_response() {
     assert_eq!(reports.len(), 0);
 }
 
+// Test a successful collect request submission.
+// This checks that the Leader reponds with the correct collect ID.
+// TODO(nakatsuka-y) Implement a test case when a Leader rejects a collect request.
+#[tokio::test]
+async fn http_post_collect_success() {
+    let leader = MockAggregator::new();
+    let task_id = leader.nominal_task_id();
+    let task_config = leader.get_task_config_for(task_id).unwrap();
+    let now = 1637361337;
+
+    // Collector: Create a CollectReq.
+    let collector_collect_req = CollectReq {
+        task_id: task_id.clone(),
+        batch_interval: task_config.current_batch_window(now),
+        agg_param: Vec::default(),
+    };
+    let req = DapRequest {
+        media_type: Some(MEDIA_TYPE_COLLECT_REQ),
+        payload: collector_collect_req.get_encoded(),
+        url: task_config.helper_url.join("/collect").unwrap(),
+        sender_auth: Some(BearerToken::from(COLLECTOR_BEARER_TOKEN.to_string())),
+    };
+
+    // Leader: Handle the CollectReq received from Collector.
+    let url = leader.http_post_collect(&req).await.unwrap();
+    let (leader_collect_id, leader_collect_req) =
+        leader.get_pending_collect_jobs(&task_id).await.unwrap()[0].clone();
+
+    // Check that the CollectReq sent by Collector is the same that is received by Leader.
+    assert_eq!(collector_collect_req, leader_collect_req);
+
+    // Check that the collect_id included in the URI is the same with the one received
+    // by Leader.
+    // TODO(nakatsuka-y) Use better way to parse the URI.
+    let parts = Url::parse(url.as_str()).expect("Unable to parse URL");
```

Comment:
I think [matchit](https://docs.rs/matchit/latest/matchit/) would be a good choice for parsing this. See this example: https://docs.rs/matchit/latest/matchit/#named-parameters.

Reply:
This works great! Thanks for suggesting.